### PR TITLE
production docker files

### PR DESCRIPTION
**Pull request by yonglbl**
the original docker-compose didn't really point to the builded react files , react/build

### CODE REVIEW

File: BRAVE/docker-compose.prod.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+version: "3.8"
```
Comment (dsclassen) [inline BRAVE/docker-compose.prod.yml:1]: New version of docker do not use `version` it has been deprecated and can be removed.

File: BRAVE/docker-compose.prod.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+services:
+  django:
+    build: ./backend
+    container_name: dev-django
```
Comment (dsclassen) [inline BRAVE/docker-compose.prod.yml:6]: if this is meant to be for "production" should the containers be named `prod-django` ?

File: BRAVE/docker-compose.prod.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      - 3001:3000
+    env_file:
+      - ./.env.dev
+    environment:
+      - CHOKIDAR_USEPOLLING=true
+    depends_on:
+      - django
+  db:
+    image: postgres:16.1-alpine
+    container_name: dev-db
```
Comment (dsclassen) [inline BRAVE/docker-compose.prod.yml:35]: `prod-db` ?

File: BRAVE/docker-compose.prod.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    ports:
+      - 5432:5432
+    env_file: ./.env.dev
+    volumes:
+      - postgres_data:/var/lib/postgresql/data
+  nginx:
+    build: ./nginx
+    container_name: dev-nginx
```
Comment (dsclassen) [inline BRAVE/docker-compose.prod.yml:43]: `prod-nginx`

File: BRAVE/docker-compose.prod.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    env_file:
+      - ./.env.dev
+      - ./backend/.env
+    volumes:
+      - ./backend/:/srv/app
+      - static:/srv/app/static
+      - media:/srv/app/media
+    depends_on:
+      - db
+  react-prod:
```
Comment (dsclassen) [inline BRAVE/docker-compose.prod.yml:19]: leave service name as `react` and rename container to `prod-react` to be consistent with naming of other services and containers.

File: BRAVE/frontend/Dockerfile.prod
Comment (dsclassen) [inline BRAVE/frontend/Dockerfile.prod:1]: node v16 is getting a bit outdated.

Maybe check if the current app will work with node v20 (current LTS)?

`FROM node:20-alpine AS build`

**Comment by dsclassen** [comment]
Can you please add `static/` to the `.gitignore` file in `backend`

**Comment by dsclassen** [comment]
I pushed a few edits needed for me to run `docker compose -f docker-compose.prod.yml build`

**Comment by dsclassen** [comment]
I still get this error:


```
=> ERROR [react-prod build 7/7] RUN npm run build                                                                                                                                                            41.4s
------                                                                                                                                                                                                              
 > [react-prod build 7/7] RUN npm run build:                                                                                                                                                                        
0.457                                                                                                                                                                                                               
0.457 > frontend@0.1.0 build                                                                                                                                                                                        
0.457 > craco build                                                                                                                                                                                                 
0.457 
1.793 Creating an optimized production build...
4.811 One of your dependencies, babel-preset-react-app, is importing the
4.811 "@babel/plugin-proposal-private-property-in-object" package without
4.811 declaring it in its dependencies. This is currently working because
4.811 "@babel/plugin-proposal-private-property-in-object" is already in your
4.811 node_modules folder for unrelated reasons, but it may break at any time.
4.811 
4.811 babel-preset-react-app is part of the create-react-app project, which
4.811 is not maintianed anymore. It is thus unlikely that this bug will
4.811 ever be fixed. Add "@babel/plugin-proposal-private-property-in-object" to
4.811 your devDependencies to work around this error. This will make this message
4.811 go away.
4.811   
41.19 Failed to compile.
41.19 
41.19 [eslint] 
41.19 src/Species.js
41.19   Line 137:28:  'viewState' is not defined  no-undef
41.19 
41.19 Search for the keywords to learn more about each error.
41.19 
41.19 
------
failed to solve: process "/bin/sh -c npm run build" did not complete successfully: exit code: 1
```

File: BRAVE/docker-compose.prod.yml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+      - db-prod
+
+  react-prod:
+    build: 
+      context: ./frontend
+      dockerfile: Dockerfile.prod
+    container_name: react-prod
+    command: npm start
+    ports:
+      - 3001:3000
```
Comment (dsclassen) [inline BRAVE/docker-compose.prod.yml:25]: This is a very minor issue, but since port `3001` is currently in use can we please change this to `3000`

**Comment by dsclassen** [comment]
Did you see my notes about `CORS` in https://github.com/lbl-cbg/t5-commons/issues/3#issuecomment-2174441876
I think we will need to configure both nginx and other components to properly pass modified headers.... or something along those lines.

**Comment by yonglbl** [comment]
will do

**Comment by dsclassen** [comment]
this seems to be working

**Review by dsclassen — APPROVED** [review]

**Comment by dsclassen** [comment]
Apparently now that I am the author of the most recent commit I am unable to "Review" the PR.

**Comment by dsclassen** [comment]
Someone please merge this into `main`

**Comment by yonglbl** [comment]
@ajtritt can you approve the merge?

**Review by ajtritt — APPROVED** [review]

**Pull request merged**